flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/ScalarFunctionsTest.scala

```diff
@@ -550,6 +550,45 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
       "1111111111111111111111111111111111111111111111111111111111111111")
   }
 
+  @Test
+  def testRegexpReplace(): Unit = {
```

Comment:
Add a test for no matching (the original string should be unchanged). Also, test some special replacement strings such as "$" and "\\".

Reply:
Flink table&sql parse "\" triggered a bug, see [FLINK-10281](https://issues.apache.org/jira/browse/FLINK-10281?filter=-1) reported by me. @twalthr is fixing and merging soon.

Reply:
@twalthr I have added test cases for many special characters, and everything works fine so far. You did a great job. 👍